cl/lib/elasticsearch_utils.py

```diff
@@ -1098,8 +1425,229 @@ def do_es_feed_query(
+                p_filters = reduce(operator.iand, parent_filters)
+                parent_query = Q(
+                    "bool",
+                    filter=p_filters,
+                    should=string_query,
+                    minimum_should_match=1,
+                )
+                q_should.append(parent_query)
+
+    if not q_should:
+        return [], join_query
+
+    return (
+        Q(
+            "bool",
+            should=q_should,
+        ),
+        join_query,
+    )
+
+
+def limit_inner_hits(
+    get_params: QueryDict, results: Page, search_type: str
+) -> None:
+    """Limit inner hits of has_child query results.
+
+    :param get_params: The user search params.
+    :param results: The Page object containing search results.
+    :param search_type: The search type to perform.
+    :return: None, the function updates the results in place.
+    """
+
+    hits_limit, _ = get_child_top_hits_limit(get_params, search_type)
+    match search_type:
+        case SEARCH_TYPES.RECAP | SEARCH_TYPES.DOCKETS:
+            child_type = "recap_document"
+        case _:
+            return
+
+    for result in results:
+        try:
+            inner_hits = [
+                hit
+                for hit in result.meta["inner_hits"][
+                    f"filter_query_inner_{child_type}"
+                ]["hits"]["hits"]
+            ]
+        except KeyError:
+            result["child_docs"] = []
+            result["child_remaining"] = False
+            continue
+
+        count_hits = len(inner_hits)
+        if count_hits > hits_limit:
+            result["child_docs"] = inner_hits[:hits_limit]
+            result["child_remaining"] = True
+        else:
+            result["child_docs"] = inner_hits
+            result["child_remaining"] = False
+
+
+def get_child_top_hits_limit(
+    search_params: QueryDict | CleanData, search_type: str
+) -> tuple[int, int]:
+    """Get the frontend and query hit limits for child documents.
+
+    :param search_params: Either a QueryDict or CleanData object containing the
+    search parameters.
+    :param search_type: Elasticsearch DSL Search object
+    :return: A two tuple containing the limit for frontend hits, the limit for
+     query hits.
+    """
+
+    frontend_hits_limit = settings.CHILD_HITS_PER_RESULT
+    # Increase the CHILD_HITS_PER_RESULT value by 1. This is done to determine
+    # whether there are more than CHILD_HITS_PER_RESULT results, which would
+    # trigger the "View Additional Results" button on the frontend.
+    query_hits_limit = settings.CHILD_HITS_PER_RESULT + 1
+    match search_type:
+        case SEARCH_TYPES.RECAP:
+            docket_id_query = re.search(
+                r"docket_id:\d+", search_params.get("q", "")
+            )
+            if docket_id_query:
+                frontend_hits_limit = settings.VIEW_MORE_CHILD_HITS
+                query_hits_limit = settings.VIEW_MORE_CHILD_HITS
```

Comment:
We should refactor this code. We're repeating the same code in both cases. I think we can refactor this structural pattern matching using IF-statements. Here's my proposal: 

```python
def get_child_top_hits_limit(
    search_params: QueryDict | CleanData, search_type: str
) -> tuple[int, int]:

  frontend_hits_limit = settings.CHILD_HITS_PER_RESULT
  query_hits_limit = settings.CHILD_HITS_PER_RESULT + 1
  
  if search_type not in [SEARCH_TYPES.RECAP, SEARCH_TYPES.DOCKETS]:
      return frontend_hits_limit, query_hits_limit
  
  if search_type == SEARCH_TYPES.DOCKETS:
      frontend_hits_limit = 1
  
  docket_id_query = re.search(r"docket_id:\d+", search_params.get("q", ""))
  if docket_id_query:
      frontend_hits_limit = settings.VIEW_MORE_CHILD_HITS
      query_hits_limit = settings.VIEW_MORE_CHILD_HITS
  
  return frontend_hits_limit, query_hits_limit
```

Reply:
Great, yeah your proposal seems good to me, thanks!